### PR TITLE
Run2 jets and gen jets

### PR DESCRIPTION
**Pull request by adewit**
Updated jet and gen jet producers + configs. The example config in ./test/miniaod_cfg_jet-example.py shows how to build these objects both from packedPFCandidates + the slimmedJets/slimmedGenJets collections in miniaod and from aod. 
Remarks:
- When producing icPFJets from a slimmedJets collection, the pu id MVA value stored in the slimmedJets collection is used. It is not jet possible to rerun the pu id mva on jets from miniAOD (even if you recreate the collection yourself), but this will be possible in CMSSW 74
- The example config doesn't show how to create pat jets. The procedure is similar to CMSSW 53X, but some of the options have changed.

### CODE REVIEW

**Comment by adewit** [comment]
please test

File: python/default_producers_cfi.py [hunk 1]
```diff
@@ -290,6 +290,48 @@
...
+icPFJetFromPatProducer = cms.EDProducer('ICPFJetFromPatProducer',
+    branch                    = cms.string("pfJetsFromPat"),
+    input                     = cms.InputTag("ak4PFJets"),
+    #### The srcConfig PSet when the input is a reco::Jet collection
+    #### The srcConfig PSet when the input is a pat::Jet collection
+     srcConfig = cms.PSet(
+       includeJetFlavour         = cms.bool(False),
+       includeJECs               = cms.bool(False),
+       inputSVInfo               = cms.InputTag("secondaryVertexTagInfosAOD"),
+       requestSVInfo             = cms.bool(False)
```
Comment (ajgilbert) [inline python/default_producers_cfi.py]: should get default `isSlimmed = cms.bool(False)` here?

File: plugins/ICGenJetProducer.cc [hunk 2]
```diff
@@ -61,7 +62,8 @@ void ICGenJetProducer::produce(edm::Event& event,
     dest.set_energy(src.energy());
     dest.set_charge(src.charge());
     dest.set_flavour(0);
-    dest.set_n_constituents(src.getGenConstituents().size());
+    if(is_slimmed_) dest.set_n_constituents(src.numberOfDaughters());
+    else dest.set_n_constituents(src.getGenConstituents().size());
```
Comment (ajgilbert) [inline plugins/ICGenJetProducer.cc]: do you know if `src.numberOfDaughters()` would give the same answer as `src.getGenConstituents()` in a normal genjet (built from reco::GenParticles)? I guess we also have a problem here that there's no way to record/request the constituent particles in miniAOD as these are built from the PackedGenParticle collection right?

Comment (adewit) [inline plugins/ICGenJetProducer.cc]: I think they would give the same answer; numberOfDaughters() points to the PackedGenParticles collection, which contains all stable gen particles.
The standard way to request constituents won't work. Links to the daughters in the PackedGenParticles collection are available and there are links between the packed and pruned gen particles, so we can do something (but maybe not as much as for normal gen jets)

**Comment by adewit** [comment]
please test

**Pull request merged**